aviatrix/resource_aviatrix_cloudn_registration.go

```diff
@@ -0,0 +1,222 @@
+package aviatrix
+
+import (
+	"context"
+	"log"
+
+	"github.com/AviatrixSystems/terraform-provider-aviatrix/v2/goaviatrix"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func resourceAviatrixCloudnRegistration() *schema.Resource {
+	return &schema.Resource{
+		CreateWithoutTimeout: resourceAviatrixCloudnRegistrationCreate,
+		ReadWithoutTimeout:   resourceAviatrixCloudnRegistrationRead,
+		UpdateWithoutTimeout: resourceAviatrixCloudnRegistrationUpdate,
+		DeleteWithoutTimeout: resourceAviatrixCloudnRegistrationDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"address": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "CloudN IP Address or FQDN",
+			},
+			"username": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "CloudN username",
+			},
+			"password": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Sensitive:   true,
+				Description: "CloudN password",
+			},
+			"name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "CloudN name to register on controller",
+			},
+			"local_as_number": {
```

Comment:
Switch local_as_number and prepend_as_path since local_as_number is optional and computed, the other one is just optional.